editor/plugins/script_editor_plugin.cpp

```diff
@@ -3222,6 +3230,10 @@ void ScriptEditor::get_window_layout(Ref<ConfigFile> p_layout) {
 	}
 
 	p_layout->set_value("ScriptEditor", "open_scripts", scripts);
+	PackedInt32Array selected_scripts = script_list->get_selected_items();
```

Comment:
Would it not be better to store the resource path, not sure it's guaranteed to load in the same order / external modification? If the file is moved / deleted we wouldn't load it here anyway.

Reply:
Do you know what's the sense behind of ScriptEditor::set_scene_root_script ? I noticed that it always resets the script to the first in the list (while the scene root hasn't even a script attached to it). Besides, in my opinion opening the scene root script is not very user-friendly.

Reply:
It's an very old feature. If I can remember correctly, it used to be quite useful as originally script and scene editing were tied together, so it would automatic open the roots node's script. Can't say how many use it now.